test/TestApps/ToastNotificationsTestApp/main.cpp

```diff
@@ -441,7 +562,141 @@ bool VerifyShowToast_Unpackaged()
         return false;
     }
 
-    // TODO: Verify the toast was posted by calling History APIs.
+    return VerifyToastIsActive(toast.ToastId());
+}
+
+bool VerifyRemoveAllAsync()
+{
+    auto toastNotificationManager = winrt::ToastNotificationManager::Default();
+
+    winrt::ToastNotification toast1{ CreateToastNotification() };
+    toastNotificationManager.ShowToast(toast1);
+
+    winrt::ToastNotification toast2{ CreateToastNotification() };
+    toastNotificationManager.ShowToast(toast2);
+
+    winrt::ToastNotification toast3{ CreateToastNotification() };
+    toastNotificationManager.ShowToast(toast3);
+
+    auto getAllAsync = toastNotificationManager.GetAllAsync();
+    if (getAllAsync.wait_for(std::chrono::seconds(300)) != winrt::Windows::Foundation::AsyncStatus::Completed)
+    {
+        getAllAsync.Cancel();
+        return false;
+    }
+
+    // At this point there at least 3 active toasts, the ones we just posted but there could also be others that are still active.
+    if (getAllAsync.get().Size() < 3)
```

Comment:
Nit: `!= 3`

Also, how other toasts could still be active?

Now I'm seeing that for packaged scenarios, in main() we Register but we don't Unregister, which to me it seems wrong. This may be the reason on which you may be seeing toasts leaked from other tests.